cdap-api-common/src/main/java/co/cask/cdap/internal/io/SchemaTypeAdapter.java

```diff
@@ -261,6 +305,36 @@ private Schema readRecord(JsonReader reader, Map<String, Schema> knownRecords) t
     return schema;
   }
 
+  private List<Schema.Field> getFields(String recordName, JsonReader reader, Map<String, Schema> knownRecords)
+    throws IOException {
+    knownRecords.put(recordName, null);
+    List<Schema.Field> fieldBuilder = new ArrayList<>();
+    reader.beginArray();
+    while (reader.peek() != JsonToken.END_ARRAY) {
+      reader.beginObject();
+      String fieldName = null;
+      Schema innerSchema = null;
+
+      while (reader.hasNext()) {
+        String name = reader.nextName();
+        switch(name) {
+          case "name":
+            fieldName = reader.nextString();
+            break;
+          case "type":
+            innerSchema = read(reader, knownRecords);
+            break;
+          default:
+            reader.skipValue();
```

Comment:
I think as long as the schema has required fields such as (`type`, `name`), we should not error out. I tried adding extra unknown fields in the avro schema json and avro parser was also able to parse it successfully.